src/api/app/controllers/webui/patchinfo_controller.rb

```diff
@@ -193,14 +193,43 @@ def save
   end
 
   def remove
+    # checks
+    if @package.name == '_project'
+      raise DeletePackageNoPermission.new '_project package can not be deleted.'
+    end
+
+    unless User.current.can_modify_package?(@package)
+      raise DeletePackageNoPermission.new "no permission to delete package #{@package.name} in project #{@project.name}"
+    end
+
+    # deny deleting if other packages use this as develpackage
+    # Note: This might raise an exception
+    @package.can_be_deleted?
```

Comment:
I wouldn't. This is has nothing to do with the permission of the user to delete this package. This is some functional requirement for the application